whitepapers/0011_accountant.md

```diff
@@ -0,0 +1,190 @@
+
+Transferring a native token across the tokenbridge (locking) increases the balance on both the custody account and wrapped account on the destination chain.  Transferring a wrapped token back across the tokenbridge (burning) and subsequently unlocking its native asset decreases the balance on both the wrapped and custody accounts.  Any transfer causing the balance of an account to become negative is invalid and gets rejected.
+
+Each account is identified via a `(chain_id, token_chain, token_address)` tuple.  Custody accounts are those where `chain_id == token_chain`, while the rest are wrapped accounts. See the [governance](#governance) section below for futher clarification.
+
+#### Governance
+
+Account balances recorded by the accountant may end up inaccurate due to bugs in the program or exploits of the token bridge.  In these cases, the guardians may issue governance actions to manually modify the balances of the accounts.  These governance actions must be signed by a quorum of guardians.
+
+#### Examples
+
+Example 1: A user sends SOL from solana to ethereum.  In this transaction the  `(Solana, Solana, SOL)` account is debited and the `(Ethereum, Solana, SOL)` account is credited, increasing the balances of both accounts.
+
+Example 2: A user sends wrapped SOL from ethereum to solana.  In this transaction the `(Ethereum, Solana, SOL)` account is debited and the `(Solana, Solana, SOL)` account is credited, decreasing the balances of both accounts.
+
+Example 3: An attacker exploits a bug in the solana contract to print fake wrapped ETH and transfers it over the bridge. This could drain real ETH from the ethereum contract. Thanks to the accounting contract, the total amount drained is limited by the total number of wrapped ETH issued by the solana token bridge contract.  Since the token bridge is still liable for all the legitimately issued wrapped ETH the account balances no longer reflect reality.
+
+Wormhole decides to cover the cost of the hack and transfers the ETH equivalent to the stolen amount to the token bridge contract on ethereum.  To update the accounts, the guardians must issue 2 governance actions: one to increase the balance of the `(Ethereum, Ethereum, ETH)` custody account and one to increase the balance of the `(Solana, Ethereum, ETH)` wrapped account.
+
+Example 4: A user sends wrapped SOL from ethereum to polygon.  Even though SOL is not native to either chain, the transaction is still recorded the same way: the `Ethereum/SOL` mint account is debited and the `Polygon/SOL` mint account is credited.  The total liability of the token bridge has not changed and instead we have just moved some liability from ethereum to polygon.
+
+#### Creating accounts
+
+Since users can send any arbitrary token over the token bridge, the accountant should automatically create accounts for any new token.  To determine whether the newly created account should be a custody account or a mint account, these steps are followed:
```

Comment:
"mint account" -> "wrapped account"

Reply:
Fixed in 5133abeb5ed1a3c648e5721b0bc69628c1d15abe